packages/amplify-graphql-conversation-transformer/src/resolvers/verify-session-owner-resolver.ts

```diff
@@ -0,0 +1,67 @@
+import { MappingTemplate } from '@aws-amplify/graphql-transformer-core';
+import { MappingTemplateProvider } from '@aws-amplify/graphql-transformer-interfaces';
+import { dedent } from 'ts-dedent';
+import { JSResolverFunctionProvider } from './js-resolver-function-provider';
+
+/**
+ * Creates a mapping template for verifying the session owner in a conversation.
+ *
+ * @returns {JSResolverFunctionProvider} An object containing request and response MappingTemplateProviders.
+ */
+export const verifySessionOwnerMappingTemplate = (): JSResolverFunctionProvider => {
+  const req = createVerifySessionOwnerRequestFunction();
+  const res = createVerifySessionOwnerResponseFunction();
+  return { req, res };
+};
+
+/**
+ * Creates the request function for verifying the session owner in a conversation.
+ *
+ * @returns {MappingTemplateProvider} A MappingTemplateProvider for the request function.
+ */
+const createVerifySessionOwnerRequestFunction = (): MappingTemplateProvider => {
+  const requestFunctionString = `
+      export function request(ctx) {
+        const { authFilter } = ctx.stash;
+
+        const query = {
+          expression: 'id = :id',
+          expressionValues: util.dynamodb.toMapValues({
+            ':id': ctx.args.conversationId
+          })
+        };
+
+        const filter = JSON.parse(util.transform.toDynamoDBFilterExpression(authFilter));
+
+        return {
+          operation: 'Query',
+          query,
+          filter
+        };
+      }
+      `;
+
+  return MappingTemplate.inlineTemplateFromString(dedent(requestFunctionString));
+};
+
+/**
+ * Creates the response function for verifying the session owner in a conversation.
+ *
+ * @returns {MappingTemplateProvider} A MappingTemplateProvider for the response function.
+ */
+const createVerifySessionOwnerResponseFunction = (): MappingTemplateProvider => {
+  const responseFunctionString = `
+      export function response(ctx) {
+        if (ctx.error) {
+          util.error(ctx.error.message, ctx.error.type);
+        }
+        if (ctx.result.items.length !== 0 && ctx.result.scannedCount === 1) {
+          return ctx.result.items[0];
```

Comment:
Since we're already applying the Auth filter in the request, can't we say that the caller is the owner without examining the `scannedCount`? I might be missing something.

Reply:
> can't we say that the caller is the owner without examining the scannedCount?

Yes, I believe so. Because we're querying the `Conversation<RouteName>` table where the `id` is the PK (no SK), as far as I can see the `scannedCount` check is unnecessary in the first if statement. I included it because that's the current behavior for response mapping template evaluation of get queries for models with owner auth.